File: 백준/Silver/1978. 소수 찾기/소수 찾기.cc
#include <iostream>

using namespace std;

int is_prime(int num);

int main()
{
    int N, num;
    int prime_count = 0;
    
    cin >> N;
    for (int i = 0; i < N; i++)
    {
        cin >> num;
        prime_count += is_prime(num);
    }
    cout << prime_count;
    
    return 0;
}

int is_prime(int num)
{
    if (num == 1)
        return 0;
    for (int i = 2; i <= num / 2; i++)
    {
        if (num % i == 0)
            return 0;
    }
    
    return 1;
}

File: 백준/Bronze/1978. 소수 찾기/소수 찾기.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.StringTokenizer;

class Main {

	public static void main(String[] args) throws IOException {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

		int n = Integer.parseInt(br.readLine());
		
		StringTokenizer st = new StringTokenizer(br.readLine());
		
		int cnt = 0;
		for (int i = 0; i < n; i++) {
			cnt += isPrime(Integer.parseInt(st.nextToken()));
		}
		
		System.out.println(cnt);
	}
	
	static int isPrime(int num) {
		if (num <= 1) return 0;
		
		for (int i = 2; i < num; i++) {
			if (num % i == 0) return 0;
		}
		return 1;
	}
}